static/src/javascripts/projects/commercial/modules/dfp/prepare-permutive.js

```diff
@@ -0,0 +1,137 @@
+// @flow
+import { loadScript } from 'lib/load-script';
+import config from 'lib/config';
+import reportError from 'lib/report-error';
+
+declare var permutive: any;
+
+/* eslint-disable */
+export const initPermutive = (): Promise<void> => {
+    return new Promise(resolve => {
+        (function (n, e, o, r, i) {
+            if (!e) {
+                (e = e || {}),
+                    (window.permutive = e),
+                    (e.q = []),
+                    (e.config = i || {}),
+                    (e.config.projectId = o),
+                    (e.config.apiKey = r),
+                    (e.config.environment =
+                        e.config.environment || 'production');
+                for (
+                    let t = [
+                            'addon',
+                            'identify',
+                            'track',
+                            'trigger',
+                            'query',
+                            'segment',
+                            'segments',
+                            'ready',
+                            'on',
+                            'once',
+                            'user',
+                            'consent',
+                        ],
+                        c = 0;
+                    c < t.length;
+                    c++
+                ) {
+                    const f = t[c];
+                    e[f] = (function(n) {
+                        return function() {
+                            const o = Array.prototype.slice.call(arguments, 0);
+                            e.q.push({ functionName: n, arguments: o });
+                        };
+                    })(f);
+                }
+            }
+        })(
+            document,
+            window.permutive,
+            'd6691a17-6fdb-4d26-85d6-b3dd27f55f08',
+            '359ba275-5edd-4756-84f8-21a24369ce0b',
+            {}
+        );
+        (window.googletag = window.googletag || {}),
+            (window.googletag.cmd = window.googletag.cmd || []),
+            window.googletag.cmd.push(() => {
+                if (
+                    window.googletag.pubads().getTargeting('permutive')
+                        .length === 0
+                ) {
+                    const g = window.localStorage.getItem('_pdfps');
+                    window.googletag
+                        .pubads()
+                        .setTargeting('permutive', g ? JSON.parse(g) : []);
+                }
+            });
+        /* eslint-enable */
+        try {
+            const {
```

Comment:
I would consider pulling this logic out into a function that takes as an argument the config object, the reportError mechanism and a reference to `permutive`. We'd then be able to write tests to make sure this is correct. We've got some fiddly logic with undefined checks, and probably want to ensure that we're sending the correct fields to permutive. It doesn't have to be in this PR (we can shoot this over to Jane and get them to check it like this, behind the 0% test) but I think this logic is important enough that it warrants tests, if only to prevent regressions.

Reply:
Good shout 👍One of the main reasons we moved it here was to be able to test it. Personally I'm ok for us to merge this (if you are) and for me to make a PR for tests so we can confirm from Permutive that this is working in the meantime.

Reply:
Sounds great to me.